### DISCUSSION TITLE
ComboBox with dynamic model

### DISCUSSION THREAD

**Discussion by nununoisy** [slint-ui/slint#2786] [Q&A] [answered]
I have a ComboBox widget in my application that is intended to allow the user to pick from sections in a file they choose. Setting the `model` of the ComboBox to an in `[string]` property works in that the choices provided update when the user picks a different file. However, if the user selects any option before the model update, the previous choice's text is not cleared from the box. This occurs on SlintPad and Windows using Slint on Rust with the Qt and winit-femtovg backends. [MRE on SlintPad][slintpad-mre]

I've tried the following to no avail:
- Binding `current-index` to an in-out `bool` property and setting it to `-1` before and after updating the model
- Binding `current-value` to an in-out `string` property and setting it to a placeholder value before and after updating the model
- Updating the model to an empty array first before updating it to the next model
  - I have also tried this with a conditional that only renders the ComboBox if the model is not empty
- Calling `request_redraw()` after updating the model

If anyone has any ideas for this they would be much appreciated.

[slintpad-mre]: https://slint-ui.com/releases/1.0.2/editor/?snippet=import+%7B+ComboBox%2C+CheckBox+%7D+from+%22std-widgets.slint%22%3B%0Aexport+component+Example+inherits+Window+%7B%0A++++width%3A+200px%3B%0A++++height%3A+130px%3B%0A++++ComboBox+%7B%0A++++++++y%3A+0px%3B%0A++++++++width%3A+self.preferred-width%3B%0A++++++++height%3A+self.preferred-height%3B%0A++++++++model%3A+check.checked+%3F+%5B%22other%22%2C+%22other+2%22%5D+%3A+%5B%22first%22%2C+%22second%22%2C+%22third%22%5D%3B%0A++++%7D%0A++++check+%3A%3D+CheckBox+%7B%0A++++++++text%3A+%22Use+second+set%22%3B%0A++++%7D%0A%7D%0A

**Comment by ogoffart** [comment, marked as answer]
You may need to create a callback that sets the text value manually when you change the model.

So something like this: https://slint-ui.com/releases/1.0.2/editor/?snippet=import+%7B+ComboBox%2C+CheckBox+%7D+from+%22std-widgets.slint%22%3B%0Aexport+component+Example+inherits+Window+%7B%0A++++width%3A+200px%3B%0A++++height%3A+130px%3B%0A++++cb+%3A%3D+ComboBox+%7B%0A++++++++y%3A+0px%3B%0A++++++++width%3A+self.preferred-width%3B%0A++++++++height%3A+self.preferred-height%3B%0A++++++++model%3A+check.checked+%3F+%5B%22other%22%2C+%22other+2%22%5D+%3A+%5B%22first%22%2C+%22second%22%2C+%22third%22%5D%3B%0A++++%7D%0A++++check+%3A%3D+CheckBox+%7B%0A++++++++text%3A+%22Use+second+set%22%3B%0A++++++++toggled+%3D%3E+%7B+cb.current-value+%3D+%22+%22%3B+%7D%0A++++%7D%0A%7D%0A

**Comment by nununoisy** [comment]
Sorry for getting back to this so late. Using a binding to `current-value` did work, it's just that I forgot to use `<=>` when linking the property. 'Twas my own inexperience that got me.
